ppr-ui/src/components/collateral/EditCollateral.vue

```diff
@@ -0,0 +1,215 @@
+                      </template>
+                    </v-select>
+                  </v-col>
+                  <v-col cols="4">
+                    <v-select
+                      :items="getYears"
+                      filled
+                      label="Year"
+                      v-model="vehicle.year"
+                      id="txt-years"
+                    >
+                    </v-select>
+                  </v-col>
+                </v-row>
+                <v-row>
+                  <v-col>
+                    <v-text-field
+                      filled
+                      label="Make"
+                      id="txt-make"
+                      v-model="vehicle.make"
+                      persistent-hint
+                    />
+                  </v-col>
+                </v-row>
+                <v-row>
+                  <v-col>
+                    <v-text-field
+                      filled
+                      label="Model"
+                      id="txt-model"
+                      v-model="vehicle.model"
+                      persistent-hint
+                    />
+                  </v-col>
+                </v-row>
+                <v-row>
+                  <v-col>
+                  <div class="form__row form__btns">
+                    <v-btn
+                      large
+                      outlined
+                      color="error"
+                      :disabled="activeIndex === -1"
+                      @click="removeVehicle()"
+                      id="remove-btn"
+                      >Remove
+                    </v-btn>
+
+                    <v-btn
+                      large
+                      id="done-btn"
+                      class="m1-auto"
+                      color="primary"
+                      @click="validateForm()"
+                    >
+                      Done
+                    </v-btn>
+
+                    <v-btn
+                      id="cancel-btn"
+                      large
+                      outlined
+                      color="primary"
+                      @click="resetFormAndData(true)"
+                    >
+                      Cancel
+                    </v-btn>
+                  </div>
+                  </v-col>
+                </v-row>
+              </v-form>
+            </div>
+          </div>
+        </li>
+      </ul>
+    </v-expand-transition>
+  </div>
+</template>
+
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  reactive,
+  toRefs
+} from '@vue/composition-api'
+import { useGetters, useActions } from 'vuex-composition-helpers'
+import { VehicleCollateralIF } from '@/interfaces' // eslint-disable-line no-unused-vars
+import { VehicleTypes } from '@/resources'
+
+export default defineComponent({
+  props: {
+    activeIndex: {
+      type: Number,
+      default: -1
+    },
+    invalidSection: {
+      type: Boolean,
+      default: false
+    }
+  },
+  emits: ['addEditVehicle', 'resetEvent'],
```

Comment:
is this used at all or is it for making it easier for us to see what emit options there are? I'm wondering if it should get added to all the composition components I made previously

Reply:
I think this is necessary for an emit in the composition api? based on my research. So I think it's used